### DISCUSSION TITLE
CPU Usage on VM with SEB Windows

### DISCUSSION THREAD

**Discussion by hg-f** [SafeExamBrowser/seb-win-refactoring#114] [Q&A]
Hi
We have recently moved from on-prem PCs to running SEB in Azure using AVD. We have now seen that a single SEB session is consuming nearly 50% of a 4 vCPU machine and about 35% of an 8 vCPU machine. This means very few users can be connected per machine, but worse, once a few users are using SEB, video within exams are basically unusable.

Is there a tweak or setting you can recommend to stop SEB from consuming so much CPU? Just starting SEB causes the spike, even before starting the exam is initialised.

Thanks

**Comment by dbuechel** [comment]
There aren't any configuration values regarding the concern you mentioned. Are you sure that everything is well configured in your virtualization infrastructure? How do your machines perform when you launch any other software comparable to SEB?

**Reply by hg-f** [reply]
It wizzes along fine. I have even gone to F8 (8vCPU/32gb RAM) machine types, and all sw runs perfectly fine, but as soon as SEB is fired up, CPU spikes and even on F8 type 3 users connect and almost 100% CPU use.

**Comment by danschlet** [comment]
We don't have any issues with a self-hosted VMware Horizon View environment. Maybe Azure AVD or the used Windows system needs to be configured differently or you're using some weird exam system which is very inefficiently implemented. Which SEB version(s) are you using? Is the same SEB setup on a physical machine showing any performance issues?

We don't have experience with Azure AVD, so you need to find out on your own what the issue is when you run SEB in that environment.

**Comment by forobliot** [comment]
Hello everybody

Do you find any solution for this issue?
We installed the new version of SEB 3.x on our virtual machines in university labs but when we launch SEB the CPU usage goes to over 60%.
With this usage, we can’t use SEB in university labs as the virtualized infrastructure would crash with simultaneous students accessing exams
Our virtualized infrastructure is Red Hat Virtualization (based on oVirt) on 3 cluster.
The virtual machines are Windows 10 LTSC, 2 CPU, RAM 4GB.

Can someone help us?
Thanks in advance

**Reply by dbuechel** [reply]
We'll try to have a look at it and track the work and any updates under #378.

**Comment by hg-f** [comment]
Hi
We couldn't get this resolved properly. Even with 8vCPU you can get 4, maybe 5 users working and then it becomes unusable. Audio exams are a no go, as audio and video goes out of sync. The app is deffo an issue, but it doesn't seem the devs are fixing this issue. At least we are both confident it is not our systems.
As a workaround, we use normal exams within VM with lots of resources, but video exams, outside of the a VDI environment using a browser. Not ideal at all, but no other way to use it reliably.
Please share if you find another fix/workaround.

**Reply by danschlet** [reply]
WE NEED A WAY TO REPRODUCE AN ISSUE TO BE ABLE TO FIX IT!

As we mentioned, it works PERFECTLY on VMware Horizon VDI. Looks like your virtualization solutions are not compatible with SEB. We would love to help solve these issues, but for that one of you guys has to set up a demo exam access for us so we can test it ourselves. No help from your side, no way for us to fix it. This is a freeware software, we need YOUR help if we're supposed to solve the issue with your environment.
